### PR TITLE
Feature/substudy website consent

### PR DESCRIPTION
**Pull request by achen2401**
https://jira.movember.com/browse/TN-2732

### CODE REVIEW

File: portal/models/tou.py [hunk 1]
```diff
@@ -7,9 +7,10 @@
 from .organization import Organization, OrgTree
 from .role import ROLE, Role
 from .user import User
-
+#TODO check if it is right to add the extra TOU type here? Migration needed then?
```
Comment (pbugni) [inline portal/models/tou.py:10]: Yes, this requires a migration, to update the enumeration in the db.  I can help w/ that later if you would like.

File: portal/views/portal.py [hunk 2]
```diff
@@ -594,12 +594,21 @@ def initial_queries():
             r = ROLE.STAFF.value if r == ROLE.STAFF_ADMIN.value else r
             role = r
     terms = get_terms(user.locale_code, org, role)
+    substudy_terms = None
+    #TODO check if this is OK?
```
Comment (pbugni) [inline portal/views/portal.py:598]: Not clear on your TODO - but in general, all imports should go at the head of the file, UNLESS that creates an include cycle preventing the app from loading.

**Review by pbugni — CHANGES_REQUESTED** [review]

File: portal/models/coredata.py [hunk 3]
```diff
@@ -440,6 +444,15 @@ def required(self, user, **kwargs):
...
         return user.has_role(ROLE.PATIENT.value)
 
+class Empro_Website_Terms_Of_UseData(TOU_core):
+    tou_type = 'EMPRO website terms of use'
+
+    def required(self, user, **kwargs):
+        if not super(self.__class__, self).required(user, **kwargs):
+            return False
+        return user.has_role(ROLE.PATIENT.value) and \
+            EMPRO_RS_ID in ResearchStudy.assigned_to(user)
```
[the remark this answers is not in the document]
Comment (achen2401) [reply portal/models/coredata.py:454]: @pbugni I am not sure on this one..not clear to me from the specs:  https://promowiki.movember.com/pages/viewpage.action?spaceKey=ISS&title=Consent+and+Authentication
@mcjustin can you comment on this one?

File: portal/models/tou.py (hunk 1 above)
Comment (achen2401) [reply portal/models/tou.py:10]: thank you, @pbugni Your help in writing the migration script is much appreciated.

**Pull request closed without merging**